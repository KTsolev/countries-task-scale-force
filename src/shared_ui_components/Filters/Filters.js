import React from 'react';
import {ReactComponent as ChveronDown} from '../../assets/chevron_down.svg';

export const Filters = (props) => {
    const { filterBy, itemsPerPage, onChange, onSelect } = props;

    return (
        <form className='filter'>
            <input
                type='text'
                placeholder='filter by name' 
                name='filterBy' 
                value={filterBy} 
                onChange={onChange} 
            />
            <label htmlFor='per_page'>Item per page</label>
            <div className='svg-holder'>
                <select 
                    name='per_page'
                    value={itemsPerPage} 
                    onChange={onSelect}
                >   
                    <option selected={itemsPerPage === 20} value={20}>20</option>
                    <option selected={itemsPerPage === 50} value={50}>50</option>
                    <option selected={itemsPerPage === 100} value={100}>100</option>
                    <option selected={itemsPerPage === 150} value={150}>150</option>
                    <option selected={itemsPerPage === 200} value={200}>200</option>
                </select> 
                <ChveronDown className='svg'/>
            </div>
        </form>
    );
}