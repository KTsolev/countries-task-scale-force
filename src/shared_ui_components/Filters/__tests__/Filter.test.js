/* eslint-disable jest/no-mocks-import */
import React from 'react';
import { Filters } from '../Filters';
import renderer, { act } from 'react-test-renderer';

const onChange = jest.fn();

const onSelect = jest.fn();

describe('Filters tests', () => {
    it('should render component without props', () => {
        const tree = renderer.create(<Filters />);
            
        expect(tree.toJSON()).toMatchSnapshot();
    });

    it('should render component with props', () => {
        const tree = renderer.create(<Filters filterBy='ard' itemsPerPage={50} />);
        
        expect(tree.root.findByType('input').props.value).toBe('ard');
        expect(tree.root.findByType('select').props.value).toBe(50);

        expect(tree.toJSON()).toMatchSnapshot();
    });

    
    it('should onChange and OnSelect', async () => {
        const tree = renderer.create(<Filters filterBy='AU' onChange={onChange} itemsPerPage={100} onSelect={onSelect} />);
        
        const input = tree.root.findByType('input');
        const select = tree.root.findByType('select');

        const mEvent = { target: { value: 'AU' } };
        const sEvent = { target: { value: 100 } };

        act(() => {
          input.props.onChange(mEvent);
        });

        act(() => {
            select.props.onChange(sEvent);
        });

        expect(onChange).toHaveBeenCalledWith(mEvent);
        expect(onSelect).toHaveBeenCalledWith(sEvent);

        expect(input.props.value).toEqual('AU');
        expect(select.props.value).toEqual(100);

        expect(tree.toJSON()).toMatchSnapshot();
    });
});