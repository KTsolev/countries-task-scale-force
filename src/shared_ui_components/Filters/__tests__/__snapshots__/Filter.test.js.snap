// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Filters tests should onChange and OnSelect 1`] = `
<form
  className="filter"
>
  <input
    name="filterBy"
    onChange={
      [MockFunction] {
        "calls": Array [
          Array [
            Object {
              "target": Object {
                "value": "AU",
              },
            },
          ],
        ],
        "results": Array [
          Object {
            "type": "return",
            "value": undefined,
          },
        ],
      }
    }
    placeholder="filter by name"
    type="text"
    value="AU"
  />
  <label
    htmlFor="per_page"
  >
    Item per page
  </label>
  <div
    className="svg-holder"
  >
    <select
      name="per_page"
      onChange={
        [MockFunction] {
          "calls": Array [
            Array [
              Object {
                "target": Object {
                  "value": 100,
                },
              },
            ],
          ],
          "results": Array [
            Object {
              "type": "return",
              "value": undefined,
            },
          ],
        }
      }
      value={100}
    >
      <option
        selected={false}
        value={20}
      >
        20
      </option>
      <option
        selected={false}
        value={50}
      >
        50
      </option>
      <option
        selected={true}
        value={100}
      >
        100
      </option>
      <option
        selected={false}
        value={150}
      >
        150
      </option>
      <option
        selected={false}
        value={200}
      >
        200
      </option>
    </select>
    <test
      className="svg"
    />
  </div>
</form>
`;

exports[`Filters tests should render component with props 1`] = `
<form
  className="filter"
>
  <input
    name="filterBy"
    placeholder="filter by name"
    type="text"
    value="ard"
  />
  <label
    htmlFor="per_page"
  >
    Item per page
  </label>
  <div
    className="svg-holder"
  >
    <select
      name="per_page"
      value={50}
    >
      <option
        selected={false}
        value={20}
      >
        20
      </option>
      <option
        selected={true}
        value={50}
      >
        50
      </option>
      <option
        selected={false}
        value={100}
      >
        100
      </option>
      <option
        selected={false}
        value={150}
      >
        150
      </option>
      <option
        selected={false}
        value={200}
      >
        200
      </option>
    </select>
    <test
      className="svg"
    />
  </div>
</form>
`;

exports[`Filters tests should render component without props 1`] = `
<form
  className="filter"
>
  <input
    name="filterBy"
    placeholder="filter by name"
    type="text"
  />
  <label
    htmlFor="per_page"
  >
    Item per page
  </label>
  <div
    className="svg-holder"
  >
    <select
      name="per_page"
    >
      <option
        selected={false}
        value={20}
      >
        20
      </option>
      <option
        selected={false}
        value={50}
      >
        50
      </option>
      <option
        selected={false}
        value={100}
      >
        100
      </option>
      <option
        selected={false}
        value={150}
      >
        150
      </option>
      <option
        selected={false}
        value={200}
      >
        200
      </option>
    </select>
    <test
      className="svg"
    />
  </div>
</form>
`;
